jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JDTServicesManager.java

```diff
@@ -165,6 +165,17 @@ public List<String> getExistingContextsFromClassPath(String uri, List<String> sn
                 classpath.add(null);
             });
         }
+

```

Comment:
I believe if there is an exception caught at line 158 then the list will be one element short. This would affect the length calculation used later. I think you need to add a null to the output list if you catch an exception.

Reply:
It's the code from before, not part of changes from here, but I agree to add null to keep the list safe when exception happens (even I doubt the chance, the ctx is pre-defined from JSON files, should be always valid)

Reply:
Benson were you going to add the `classpath.add(null);` in the catch?

Reply:
no, it's better to keep that one out of this PR for better history tracing. You can open up a separate issue for your found